src/adapter/src/rbac.rs

```diff
@@ -419,3 +421,58 @@ fn generate_required_ownership(plan: &Plan) -> Vec<Ownership> {
         Plan::AlterOwner(plan) => vec![Ownership(plan.id.clone())],
     }
 }
+
+pub(crate) const fn all_object_privileges(object_type: ObjectType) -> AclMode {
+    match object_type {
+        ObjectType::Table => AclMode::INSERT
+            .union(AclMode::SELECT)
+            .union(AclMode::UPDATE)
+            .union(AclMode::DELETE),
+        ObjectType::View => AclMode::SELECT,
+        ObjectType::MaterializedView => AclMode::SELECT,
+        ObjectType::Source => AclMode::SELECT,
+        ObjectType::Sink => AclMode::empty(),
+        ObjectType::Index => AclMode::empty(),
+        ObjectType::Type => AclMode::USAGE,
+        ObjectType::Role => AclMode::empty(),
+        ObjectType::Cluster => AclMode::USAGE.union(AclMode::CREATE),
```

Comment:
Is it worth making USAGE union CREATE some rust const or Lazy?

Reply:
`union` is a `const` function, so `USAGE.union(CREATE)` should be completely evaluated at compile time.

@parker-timmerman does that sound right?

Reply:
I believe in this case we'll only evaluate `USAGE.union(CREATE)` at compile time, if `all_object_privileges` is itself called in a const context, e.g. `const X: AclMode = all_object_privileges(...)`. If it's called as part of a non-const function, then it will not. For example, in this [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ea366c3eaa620b067c58aadaa8050f62) we should fail to compile, if we were evaluating the const at compile time, but we're panicking at runtime, which indicates we're not.

If it's important that we're evaluating the union at compile time, then at the top of this function I'd add something like:
```
const USAGE_UNION_CREATE: AclMode = USAGE.union(CREATE);
```
which will for sure get evaluated at compile time.

Reply:
I would have thought that the Rust compiler has all the information it needs to rewrite the implementation of `all_object_privileges` to replace all occurrences of `AclMode::USAGE.union(AclMode::CREATE)` with it's result. 

For example, we know at compile time that `AclMode::USAGE.union(AclMode::CREATE)` is equivalent to `(1 << 8 | 1 << 9)` which can be fully evaluated at compile time. So it would make sense for the compiler to make this replacement during compilation.

I tried making a dummy example and looking at the compiled ASM, and this doesn't seem to be what's happening: https://godbolt.org/z/aKjcr8Ydr

However if I add a const variable in the function and then use that, it is replaced during compile time: https://godbolt.org/z/zMb5szj1x

So I made the separate consts and used those.

Reply:
Yeah ideally the compiler would be able to do what you're describing, it has all the info it needs, but unfortunately const eval in Rust is still very much MVP.